### DISCUSSION TITLE
Using Current Route In Shared Components

### DISCUSSION THREAD

**Discussion by fweth** [withastro/roadmap#339] [Proposal]
I don't know the best way to implement this, but it would be cool if shared components, like a header, could access the current route and import files relative to that route.  Typically I want to have breadcrumbs and siblings (pages in the same folder) linked from the header menu.  Is there some elegant solution for this?

**Comment by jasikpark** [comment]
an inelegant solution would be to do 
```astro
const PageTitle = await import(`${Astro.url.pathname}/PageTitle.astro`)
```
tho that feels like a bad idea

**Reply by fweth** [reply]
Ah, but it looks pretty much like what I wanted!  I could also do transform the `pathname` to a `Astro.glob` query to get all sibling pages :)

**Reply by fweth** [reply]
But why do you think it's a bad idea?

**Comment by florian-lefebvre** [comment]
Closing because this is outdated
